source/common/http/header_utility.h

```diff
@@ -18,6 +18,27 @@ class HeaderUtility {
 public:
   enum class HeaderMatchType { Value, Regex, Range, Present, Prefix, Suffix };
 
+  // Get all instances of the header key specified, and return the values in the
```

Comment:
Nit: Could you turn this into
```
/*
 * Get all instances of the header key specified.
 * @param ...
 * @param ...
 */
 ```
style comment and move it after `struct HeaderData`?